src/Compilers/CSharp/Portable/Compilation/MemberSemanticModel.cs

```diff
@@ -961,6 +961,65 @@ public override SyntaxTree SyntaxTree
 
         internal override IOperation GetOperationWorker(CSharpSyntaxNode node, GetOperationOptions options, CancellationToken cancellationToken)
         {
+            var bindingRoot = GetBindingRootOrInitializer(node);
+
+            var statementOrRootOperation = GetStatementOrRootOperation(bindingRoot, options, cancellationToken);
+            if (statementOrRootOperation == null)
+            {
+                return null;
+            }
+
+            // we might optimize it later
+            return statementOrRootOperation.DescendantsAndSelf().FirstOrDefault(o => !o.IsImplicit && o.Syntax == node);
```

Comment:
>statementOrRootOperation.DescendantsAndSelf().FirstOrDefault(o => !o.IsImplicit && o.Syntax == node); [](start = 19, length = 101)

Does this give us an FirstOrDefault in evaluation order? I think it would make sense to use evaluation order for this search.

Reply:
currently it will traverse IOperation in pre order and return first match. when you say evaluation order, you meant argument evaluation order? or are you saying Descendants (which uses Children property of IOperation) should traverse the tree in evaluation order?

Reply:
> currently it will traverse IOperation in pre order and return first match. when you say evaluation order, you meant argument evaluation order? or are you saying Descendants (which uses Children property of IOperation) should traverse the tree in evaluation order?

I think it is a separate discussion how Descendants should behave. For the purpose of this search, I believe we should be looking for the bottom-most "explicit" node associated with the syntax. It is hard to define in short sentence what the bottom-most mean for all possible scenarios. Probably a topic for an offline discussion. It feels like an evaluation order is easy to understand and might just give us the same result. I do not think this is blocking for this PR, but we need to figure out precise rules for this search.